crates/rmf_site_editor/src/site/load.rs

```diff
@@ -542,77 +543,58 @@ fn generate_site_entities(
         consider_id(*scenario_id);
 
         // Spawn instance modifier entities
-        let mut scenario_modifiers: ScenarioModifiers<Entity> = ScenarioModifiers::default();
-        for (instance_id, instance) in scenario_data.instances.iter() {
+        for (instance_id, instance_modifier) in scenario_data.instances.iter() {
             if let Some(instance_entity) = id_to_entity.get(&instance_id) {
```

Comment:
It seems that this `if let Some(_)` will just quietly move on if the instance is missing. If the instance is missing then the file is malformed, and we should bring that up as an error.

To easily produce a site loading error for a missing instance, you can do
```rust
let instance_entity = id_to_entity.get(&instance_id).ok_or(instance_id).for_site(site_id)?;
```

Reply:
c2e5169